Bitcoin Core 28.0

achow101 opened the pull request:
Uses the current (2024-09-30) draft release notes.

Targeting final release date of 2024-10-02

achow101 commented:
Updated release notes to [eab8c9df5996bfe0d0997f747eefbdb745381aa0](https://github.com/bitcoin-core/bitcoin-devwiki/wiki/28.0-Release-Notes-Draft/eab8c9df5996bfe0d0997f747eefbdb745381aa0).

stickies-v commented on _includes/posts/maintenance-table.md:
```diff
@@ -1,7 +1,7 @@
-| 28.x   | TBA*       | after v30.0 | after v31.0  |
+| 28.x   | 2024-10-02 | after v30.0 | after v31.0  |
```
nit: for consistency, would keep the current format?
```suggestion
| 29.x   | TBA*       | after v31.0 | after v32.0  |
| 28.x   | 2024-10-02 | after v30.0 | after v31.0  |
```

achow101 replied:
Done

stickies-v reviewed (commented):
ACK 0d81ea5353f53e522688b6c02f7354c5a5990ff8 modulo 1) adding the magnet link and 2) uploading the binaries

I've verified that:
- the release notes match https://github.com/bitcoin/bitcoin/pull/30959
- the RPC docs are an identical match with my own output
- the RPC doc changes vs 27.0 seem fine at quick glance. There's more turnover than usual, but that's mostly because of the JSON-RPC 2.0 and content-type updates, e.g.:

<details>
<summary>cd _doc/en && diff -I '^permalink.*' -I '^btcversion.*' -r 28.0.0/ 27.0.0/</summary>

```
...
diff -I ^permalink.* -I ^btcversion.* -r 28.0.0/rpc/blockchain/getblockhash.html 27.0.0/rpc/blockchain/getblockhash.html
20c20
< > curl --user myusername --data-binary '{"jsonrpc": "2.0", "id": "curltest", "method": "getblockhash", "params": [1000]}' -H 'content-type: application/json' http://127.0.0.1:8332/
---
> > curl --user myusername --data-binary '{"jsonrpc": "1.0", "id": "curltest", "method": "getblockhash", "params": [1000]}' -H 'content-type: text/plain;' http://127.0.0.1:8332/
diff -I ^permalink.* -I ^btcversion.* -r 28.0.0/rpc/blockchain/getblockheader.html 27.0.0/rpc/blockchain/getblockheader.html
41c41
< > curl --user myusername --data-binary '{"jsonrpc": "2.0", "id": "curltest", "method": "getblockheader", "params": ["00000000c937983704a73af28acdec37b049d214adbda81d7e2a3dd146f6ed09"]}' -H 'content-type: application/json' http://127.0.0.1:8332/
---
> > curl --user myusername --data-binary '{"jsonrpc": "1.0", "id": "curltest", "method": "getblockheader", "params": ["00000000c937983704a73af28acdec37b049d214adbda81d7e2a3dd146f6ed09"]}' -H 'content-type: text/plain;' http://127.0.0.1:8332/
diff -I ^permalink.* -I ^btcversion.* -r 28.0.0/rpc/blockchain/getblockstats.html 27.0.0/rpc/blockchain/getblockstats.html
66,67c66,67
< > curl --user myusername --data-binary '{"jsonrpc": "2.0", "id": "curltest", "method": "getblockstats", "params": ["00000000c937983704a73af28acdec37b049d214adbda81d7e2a3dd146f6ed09", ["minfeerate","avgfeerate"]]}' -H 'content-type: application/json' http://127.0.0.1:8332/
< > curl --user myusername --data-binary '{"jsonrpc": "2.0", "id": "curltest", "method": "getblockstats", "params": [1000, ["minfeerate","avgfeerate"]]}' -H 'content-type: application/json' http://127.0.0.1:8332/
---
> > curl --user myusername --data-binary '{"jsonrpc": "1.0", "id": "curltest", "method": "getblockstats", "params": ["00000000c937983704a73af28acdec37b049d214adbda81d7e2a3dd146f6ed09", ["minfeerate","avgfeerate"]]}' -H 'content-type: text/plain;' http://127.0.0.1:8332/
> > curl --user myusername --data-binary '{"jsonrpc": "1.0", "id": "curltest", "method": "getblockstats", "params": [1000, ["minfeerate","avgfeerate"]]}' -H 'content-type: text/plain;' http://127.0.0.1:8332/
...
```
</details>

achow101 commented on _releases/28.0.md:
```diff
@@ -0,0 +1,390 @@
+
+While the intention is to phase out support for Testnet3 in an upcoming
+version, support for it is still available via the known options in this
+release.
```
Done

achow101 commented:
Updated release notes to [24bcdbf3cd16c8a9dfce26f5f1c9b8d7ac387d8f](https://github.com/bitcoin-core/bitcoin-devwiki/wiki/28.0-Release-Notes-Draft/24bcdbf3cd16c8a9dfce26f5f1c9b8d7ac387d8f) which has additional text regarding the jsonrpc issue, and a few more PRs.

The bins are uploaded now, so the torrent magnet has been added too.

achow101 commented:
Added a commit to remove powerpc64le from the downloads since we aren't building those currently. That should fix CI.

Also added a release note for that.

Sjors commented on _releases/28.0.md:
```diff
@@ -0,0 +1,405 @@
+## and run: transmission-show -m <torrent file>
+#
+## Link should be enclosed in quotes and start with: "magnet:?
+optional_magnetlink: magnet:?xt=urn:btih:e18e92024fc9d4026cf8cdef174f03c24080fd1f&dn=bitcoin-core-28.0&tr=udp%3A%2F%2Ftracker.coppersurfer.tk%3A6969%2Fannounce&tr=udp%3A%2F%2Ftracker.openbittorrent.com%3A80&tr=udp%3A%2F%2Ftracker.opentrackr.org%3A1337%2Fannounce&tr=udp%3A%2F%2Ftracker.torrent.eu.org%3A451%2Fannounce&tr=udp%3A%2F%2Ftracker.leechers-paradise.org%3A6969%2Fannounce&tr=udp%3A%2F%2Fexp ...
```
I whitelisted it on my tracker and I'm seeding it.

stickies-v commented on _includes/templates/download.html:
```diff
@@ -90,8 +90,10 @@ <h2>{{ page.latestversion }} {{CURRENT_RELEASE}} <a type="application/rss+xml" h
         <span>
           <a href="{{ PATH_PREFIX }}/{{ FILE_PREFIX }}-{{ site.data.binaries.ppc64le }}.tar.gz" class="dl">PPC64 Linux</a>
             <span><a href="{{ PATH_PREFIX }}/{{ FILE_PREFIX }}-{{ site.data.binaries.ppc64 }}.tar.gz" class="dl" id="lin64ppc">64 bit</a>
+<!--
             -
             <a href="{{ PATH_PREFIX }}/{{ FILE_PREFIX }}-{{ site.data.binaries.ppc64le }}.tar.gz" class="dl" id="lin64ppcle">64 bit LE</a></span>
+-->
```
Dangling reference to `ppc64le` here is breaking CI. Since there's no longer the choice between two options, I would just update the "header" to `ppc64` and remove the two "options" altogether, instead of just the `ppc64le` one.
```suggestion
          <a href="{{ PATH_PREFIX }}/{{ FILE_PREFIX }}-{{ site.data.binaries.ppc64 }}.tar.gz" class="dl">PPC64 Linux</a>
<!--
            <span><a href="{{ PATH_PREFIX }}/{{ FILE_PREFIX }}-{{ site.data.binaries.ppc64 }}.tar.gz" class="dl" id="lin64ppc">64 bit</a>
            -
            <a href="{{ PATH_PREFIX }}/{{ FILE_PREFIX }}-{{ site.data.binaries.ppc64le }}.tar.gz" class="dl" id="lin64ppcle">64 bit LE</a></span>
-->
```

achow101 replied:
Done

stickies-v reviewed (commented):
Reviewed 20e0ca7a1400e78291c432cb280ad92019ad5c27:
- verified magnet link & files
- binaries uploaded
- outstanding review comments addressed

I think https://github.com/bitcoin-core/bitcoin-devwiki/wiki/28.0-Release-Notes-Draft/a944af8192ab65a6db8d8feca474e5eb842dc7d0 should be pulled in but besides the CI failure issue I think this is RFM.

achow101 commented:
> I think https://github.com/bitcoin-core/bitcoin-devwiki/wiki/28.0-Release-Notes-Draft/a944af8192ab65a6db8d8feca474e5eb842dc7d0

Done

stickies-v commented on _includes/templates/download.html:
```diff
@@ -88,10 +88,12 @@ <h2>{{ page.latestversion }} {{CURRENT_RELEASE}} <a type="application/rss+xml" h
+<!--
             <span><a href="{{ PATH_PREFIX }}/{{ FILE_PREFIX }}-{{ site.data.binaries.ppc64 }}.tar.gz" class="dl" id="lin64ppc">64 bit</a>
             -
             <a href="{{ PATH_PREFIX }}/{{ FILE_PREFIX }}-{{ site.data.binaries.ppc64le }}.tar.gz" class="dl" id="lin64ppcle">64 bit LE</a></span>
```
It seems like htmlproofer is not ignoring the commented-out elements, so perhaps best to just remove these lines entirely? We can always git revert later.

achow101 replied:
Yeah, deleted them and added a comment instead.

stickies-v reviewed (approved):
ACK 99bbc5683ed4d902b2a3b4bedf4b2fb6dbf5eeac

stickies-v commented:
ACK bed03bd5041f9a8039ec852f0aeccb8818f2a8c2

glozow commented on _includes/templates/download.html:
```diff
@@ -88,10 +88,9 @@ <h2>{{ page.latestversion }} {{CURRENT_RELEASE}} <a type="application/rss+xml" h
-            <a href="{{ PATH_PREFIX }}/{{ FILE_PREFIX }}-{{ site.data.binaries.ppc64le }}.tar.gz" class="dl" id="lin64ppcle">64 bit LE</a></span>
+          <a href="{{ PATH_PREFIX }}/{{ FILE_PREFIX }}-{{ site.data.binaries.ppc64 }}.tar.gz" class="dl">PPC64 Linux</a>
+            <span><a href="{{ PATH_PREFIX }}/{{ FILE_PREFIX }}-{{ site.data.binaries.ppc64 }}.tar.gz" class="dl" id="lin64ppc">64 bit</a></span>
+            <!-- PPC64LE Disabled for now due to guix build issue, removed from template as the binary availability check in CI doesn't ignore comments -->
```
here's how it displays for me
<img width="386" alt="image" src="https://github.com/user-attachments/assets/0af26181-2a9c-4958-bd16-a19d7cf9a149">

glozow reviewed (commented):
ACK bed03bd5041f9a8039ec852f0aeccb8818f2a8c2

light text review + built locally. didn't diff RPC docs but they lgtm
